site/src/pages/WorkspacePage/WorkspaceScheduleControls.test.tsx

```diff
@@ -0,0 +1,189 @@
+import { render, screen } from "@testing-library/react";
+import { ThemeProvider } from "contexts/ThemeProvider";
+import { QueryClient, QueryClientProvider, useQuery } from "react-query";
+import { MockWorkspace } from "testHelpers/entities";
+import { WorkspaceScheduleControls } from "./WorkspaceScheduleControls";
+import { workspaceByOwnerAndName } from "api/queries/workspaces";
+import { RouterProvider, createMemoryRouter } from "react-router-dom";
+import userEvent from "@testing-library/user-event";
+import { server } from "testHelpers/server";
+import { rest } from "msw";
+import dayjs from "dayjs";
+import * as API from "api/api";
+import { GlobalSnackbar } from "components/GlobalSnackbar/GlobalSnackbar";
+
+const Wrapper = () => {
+  const { data: workspace } = useQuery(
+    workspaceByOwnerAndName(MockWorkspace.owner_name, MockWorkspace.name),
+  );
+
+  if (!workspace) {
+    return null;
+  }
+
+  return <WorkspaceScheduleControls workspace={workspace} canUpdateSchedule />;
+};
+
+test("add 3 hours to deadline", async () => {
+  const user = userEvent.setup();
+  const baseDeadline = dayjs().add(3, "hour");
+  const updateDeadlineSpy = jest
+    .spyOn(API, "putWorkspaceExtension")
+    .mockResolvedValue();
+  server.use(
+    rest.get(
+      "/api/v2/users/:username/workspace/:workspaceName",
+      (req, res, ctx) => {
+        return res(
+          ctx.status(200),
+          ctx.json({
+            ...MockWorkspace,
+            latest_build: {
+              ...MockWorkspace.latest_build,
+              deadline: baseDeadline.toISOString(),
+            },
+          }),
+        );
+      },
+    ),
+  );
+  render(
+    <ThemeProvider>
+      <QueryClientProvider client={new QueryClient()}>
```

Comment:
Would it be better to use a mocked-out, offline-focused version of the query client? We haven't exported it out yet, but there is a function for making one in the `renderHelpers` file

Reply:
I had a few issues using it for some reason. Not sure if it was because of the cache option but I plan to revisit our tests, specifically mocks, at a diff moment.